### ISSUE TITLE
Question on instructions

### ISSUE THREAD

**Issue by brahmafear** [valkjsaaa/switchmate-firmwares#1]
Attempting to load 2.32.3 on a "light" model.  

From the iOS readme instructions, there is a little confusion as to getting the firmware file in to NRF Connect.  One process I am trying it to use Safari to browse to the github repo and correct zip file.  I can then click to download.  I can then use the "bow with up arrow" icon to send the download to NRF Connect.  Is this the preferred method?

My real question is -- do I need to unzip the firmware first?  It seems to have a .json, a .dat, and a .bin inside.  I don't want to use NRF connect to upload a ZIP if it is expecting a BIN file.  

Thanks for posting these.  Hoping to use this switchmate with one of the rasppi3 software packages to allow control from HomeAssistant.

**Comment by valkjsaaa** [comment]
You do not need to unzip the zip file. Just use the "Open in..." to open it in NRF Connect.

**Comment by brahmafear** [comment]
Got it.  Still not working for me.  I can get all the way to the step "Press start to update".  I press the "play" button at bottom.  Text changes from "Ready" to "Connecting" to "Start" within a few seconds.  No graph is ever shown in the upper window (as seen in your screenshots).  After about 30 seconds, I can hear the switchmate cycle.  But the progress percentage never moves from 0%.  If I open the SwitchMate app, I get a popup saying a device in recovery mode has been detected.  

Any ideas?  I'm ok with experimenting and possibly bricking this device.  If it isn't controllable from 3rd party software, it isn't usable to me.  Thanks in advance.

**Comment by valkjsaaa** [comment]
Hmm. I think you have done nothing wrong. I'm not sure why it won't update. Have you tried to bring the Switchmate close to your iPhone? Also, if it disconnects, you can press the "play" button again to make the app reconnect to the Switchmate. See if that will help.

**Comment by valkjsaaa** [comment]
Also, the "light" module should be the wider one without motion sensor. Make sure you have got the firmware corresponding to your model of Switchmate.

**Comment by kevingsmeg** [comment]
I also need the “Light” version of the firmware. I can load any of the 2.99 versions successfully, but none of the earlier versions. I am wondering if there are any other Light firmware that exists prior to 2.99. Thoughts?

**Comment by valkjsaaa** [comment]
[2.21](/valkjsaaa/switchmate-firmwares/blob/master/firmwares/light-2.21.zip) should be earlier than 2.99. Your library of choice do not support any of those 2.99.x firmwares?

**Comment by kevingsmeg** [comment]
True, but neither of the two earlier versions will load. All of the 2.99 versions will load. Not sure why that is, but I need a versions earlier than 2.99 to function with my Wink controller.

**Comment by cgigate** [comment]
Not able to downgrade the firmware to the swtichmate light (on 2.99.18), want to downgrade, test all versions, no success.
the "nRF Connect" connected the device, and "start" the process, but stays at 0% forever.
Use another phone, open with switchmate app, it immediately found the device "in recovery mode", and the switchmate app tries to recover flash to version 2.32.3, but not success.
Take off battery of the light, it boots up normally, still in 2.99.18
